src/api/internal.rs

```diff
@@ -605,30 +605,6 @@ impl<T: Pixel> ContextInner<T> {
       .unwrap();
     }
 
-    // Do not modify the next output frame's FrameInvariants.
```

Comment:
Why this?

Reply:
Basically this prevents us from ever modifying the "current" fi. But if we want to run with zero latency, we *must* modify the current fi, to put in its lookahead mvs etc.

Reply:
FWIW I don't understand why this condition was here in the first place.

Reply:
the tests are currently failing though.

Reply:
So, as far as I remember, this condition was here because the current fi contains the reconstruction buffer filled with reconstructed frames, but for lookahead we set the reconstruction buffer to original frames (since we need it to compute MVs). Overwriting the current frame's reconstruction buffer might break everything that needs the correct reconstructed frame.

Reply:
Right, so what I want to change this to is to these two steps happening sequentially in one receive_packet. As per the comment on rec_buffer we end up rewriting this when starting the final encode, so I don't think we actually need to avoid writing it here.

Reply:
Should be good then (especially if breaking the reconstruction buffer is signalled by some tests).

Reply:
I don't think we have any tests for this. Maybe I can add a flag in fi just for signaling this condition for assert / debugging purposes.

Reply:
Shouldn't breaking the reconstruction buffer break sync tests?

Reply:
Yes, maybe that's enough.